src/mut/soma.cpp

```diff
@@ -38,6 +38,23 @@ float Soma::surface() const
         points());
 }
 
+float Soma::maxDistance() const
+{
+    const auto ps = points();
+    if (ps.size() == 0) {
```

Comment:
I wonder if we can do something like:
```
    const auto c = center();
    return std::accumulate(begin(ps), end(ps),
        0.f,
        [](T a, T b){return std::max(a, distance(c, b));});
```

Reply:
+1. I would use `{}` instead of `0.f` though

Reply:
Also I would use `ps.empty()` instead of `ps.size() == 0`.

Reply:
Yeah, {} is more generic.
AFAIK, the ps.size() check isn't needed w/ the above code - if std::accumulate is given an empty itr, it returns the initial value.

I should have been more clear:
```

float Soma::maxDistance() const
{
    const auto ps = points();
    const auto c = center();
    return std::accumulate(begin(ps), end(ps),
        {},
        [](float a, float b){return std::max(a, distance(c, b));});
}
```

Reply:
Sounds like an abuse of std::accumulate, but should work. I was thinking about `std::max_element` before, and that calculates too much.

Reply:
since the std library doesn't have `reduce` until c++17, std::accumulate has been used instead for a long time - so it's a 'standard' usage /grumpy old man